pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -1083,4 +1117,103 @@ private void createPendingLoadTopic() {
         }
 
     }
+    
+    /**
+     * Adds given dispatcher's unackMessage count to broker-unack message count and if it reaches to the
+     * {@link #maxUnackedMessages} then it blocks all the dispatchers which has unack-messages higher than
+     * {@link #maxUnackedMsgsPerDispatcher}. It unblocks all dispatchers once broker-unack message counts decreased to
+     * ({@link #maxUnackedMessages}/2)
+     * 
+     * @param dispatcher
+     * @param numberOfMessages
+     */
+    public void addUnAckedMessages(PersistentDispatcherMultipleConsumers dispatcher, int numberOfMessages) {
+        // don't block dispatchers if maxUnackedMessages = 0
+        if (maxUnackedMessages <= 0) {
+            return;
+        }
+        int unAckedMessages = TOTAL_UNACKED_MESSAGES_UPDATER.addAndGet(this, numberOfMessages);
```

Comment:
> Then, periodically, we could check the sum() of the adder and take action if needed.

sure, but I think only problem is we want to maintain and trigger this logic on message-dispatch to make sure that blocking and unblocking works with expected numbers. With periodic check the behavior will be little unexpected. I will check it to fix it.